ledger-core/inc/core/database/Migrations.hpp

```diff
@@ -102,15 +102,15 @@ namespace ledger {
                   // we’re in sync with the database; perform the rollback normally
                   rollback<version, T>(sql);
 
-                  if (currentVersion >= 0) {
-                      // after rolling back this migration, we won’t have anything left, so we only
-                      // update the version for > 0
-                      if (currentVersion != 0) {
-                          auto prevVersion = version - 1;
-                          sql << "UPDATE __database_meta__ SET id = :id, version = :version", soci::use(T::COIN_ID), soci::use(prevVersion);
-                      }
+                  if (currentVersion > 0) {
+                      // if the migration number is greater than 0, we just perform the regular migration
+                      auto prevVersion = version - 1;
```

Comment:
You can use `constexpr` and use it as non-templated type : `Migration<prevVersion, T>`